solver/src/solver/matcha_solver/api.rs

```diff
@@ -0,0 +1,227 @@
+//! Matcha HTTP API client implementation.
+//!
+//! For more information on the HTTP API, consult:
+//! <https://0x.org/docs/api#request-1>
+//! <https://api.0x.org/>
+
+use crate::solver::solver_utils::{
+    deserialize_decimal_f64, deserialize_decimal_u256, deserialize_prefixed_hex, Slippage,
+};
+
+use anyhow::Result;
+use ethcontract::{H160, U256};
+use reqwest::{Client, IntoUrl, Url};
+use serde::Deserialize;
+use shared::http::default_http_client;
+
+/// A Matcha API quote query parameters.
+///
+/// These parameters are currently incomplete, and missing parameters can be
+/// added incrementally as needed.
+#[derive(Clone, Debug, Default)]
+pub struct SwapQuery {
```

Comment:
Should we set an `affiliateAddress` just in case? So we can track our affiliation with them?

Reply:
This costs more gas and I don't see a direct benefit. Though it might change in the future: 
http://help.matcha.xyz/en/articles/4217019-does-matcha-have-an-affiliate-program

If I am doing it, I would use the solver's address, I guess

Reply:
> This costs more gas

Does it cost more gas outside of the 20 bytes non-zero calldata? This would be a very marginal overhead. On the other hand, we can argue that since our orders are always executed in a specific manner we can compute our affiliation with Dune later on as well.

Reply:
>Does it cost more gas outside of the 20 bytes non-zero calldata? 

No, but I thought I would not use any overhead, if there is no benefit. Anyways, I can set it as you guys want...